src/avram/callbacks.cr

```diff
@@ -80,6 +104,55 @@ module Avram::Callbacks
     end
   end
 
+  macro after_run(method_name)
+    def after_run(object)
+      {% if @type.methods.map(&.name).includes?(:after_run.id) %}
+        previous_def
+      {% else %}
+        super
+      {% end %}
+
+      {{ method_name.id }}(object)
+    end
+  end
+
+  # Run the given block after the operation runs
+  #
+  # The return value from `run` will be passed to this block.
+  #
+  # ```
+  # class GenerateReport < Avram::Operation
+  #   after_run do |value|
+  #     value == "some report"
+  #   end
+  #
+  #   def run
+  #     "some report"
+  #   end
+  # end
+  # ```
+  macro after_run(&block)
+    {%
+      if block.args.size != 1
+        raise <<-ERR
+        The 'after_run' callback requires only 1 block arg to be passed.
```

Comment:
"only" sounds like it assumes that the error happened because the block given had multiple parameters but they could also have given a block with no params and this error would be worded awkwardly.

Reply:
Maybe `The 'after_run' callback requires exactly 1 block arg to be passed.`?